pkg/dynamicinstrumentation/ditypes/analysis.go

```diff
@@ -56,6 +56,21 @@ const (
 	CaptureDepthReached                             // CaptureDepthReached means the parameter wasn't captures because the data type has too many levels
 )
 
+func (r NotCaptureReason) String() string {
```

Comment:
What's the point of having `NotCaptureReason uint8` and not `NotCaptureReason string`?

Reply:
It's a good point. Memory efficiency and type safety are generally benefits to uint8 but not very relevant in this case. I'm going to leave it for consistency for now.